File: 118.cpp
// 118. Pascal's Triangle
#include <bits/stdc++.h>
using namespace std;

class Solution{
public:
    vector<int> generateRow(int row) {
        long long result = 1;
        vector<int> resRow;
        resRow.push_back(1);

        for(int col = 1; col < row; col++) {
            result = result * (row - col) / col;
            resRow.push_back(result);
        }

        return resRow;
    }

    vector<vector<int>> generate(int numRows){
        vector<vector<int>> result;

        for(int row = 0; row < numRows; row++) {
            result.push_back(generateRow(row + 1));
        }

        return result;
    }
};

int main(){

    return 0;
}

File: 118.py
# 118. Pascal's Triangle
from typing import List

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        res = [[1]]

        for i in range(numRows - 1):
            prev = [0] + res[-1] + [0]
            row = []
            for e in range(len(res[-1])):
                row.append(prev[i] + prev[i+1])
            
            res.append(row)

        return res
